Restfull API/api/tests/test_auth.py

```diff
@@ -0,0 +1,91 @@
+import unittest
+import os
+import json
+import sys
+import inspect
+
+currentdir = os.path.dirname(os.path.abspath(
+  inspect.getfile(inspect.currentframe())))
+parentdir = os.path.dirname(currentdir)
+sys.path.insert(0, parentdir)
+
+from app import create_app
+
+
+class UserAuthTestCase(unittest.TestCase):
+    
+    """This class initializes the app with test data"""
+    def setUp(self):
+        self.app = create_app(config_name="testing")
+        self.app = self.app.test_client()
+        self.app.post("/auth/register",
+                    data=json.dumps(dict(email="jeff@gmail.com",username="jefftest",
+                                password="jeffpass")), content_type="application/json")
+
+        self.app.post("/auth/login",
+                        data=json.dumps(dict(email="jeff@gmail.com",password="jeffpass")),
+                                         content_type="application/json")
+
+    def test_user_registration(self):
+        """ Test API can register a user"""
+        response = self.app.post("/auth/register",
+                        data=json.dumps(dict(email="jeff2@gmail.com",username="jefftest2",
+                            password="jeffpass2")), content_type="application/json")
+
+        self.assertEqual(response.status_code, 201)
+        response_msg = json.loads(response.data.decode("UTF-8"))
+        self.assertIn("created", response_msg["message"])
+
+
+    def test_user_already_registered(self):
+        """ Test API can check a registered user"""
+        response = self.app.post("/auth/register",
+                        data=json.dumps(dict(email="jeff@gmail.com",username="jefftest",
+                            password="jeffpass")), content_type="application/json")
+
+        self.assertEqual(response.status_code, 404)
+        response_msg = json.loads(response.data.decode("UTF-8"))
+        self.assertIn("instead", response_msg["message"])
+
+
+    def test_user_login(self):
+        """ Test API can login a user"""
+        response = self.app.post("/auth/login",
+                        data=json.dumps(dict(email="jeff@gmail.com",password="jeffpass")),
+                                         content_type="application/json")
+
+        self.assertEqual(response.status_code, 200)
+        response_msg = json.loads(response.data.decode("UTF-8"))
+        self.assertIn("valid", response_msg["message"])  
+
+
+    def test_user_login_with_wrong_password_or_username(self):
+        """ Test API can check if user used wrong password"""
+        response = self.app.post("/auth/login",
+                        data=json.dumps(dict(email="jeff@gmail.com",password="jeffpassxxx")),
+                                         content_type="application/json")
+
+        self.assertEqual(response.status_code, 404)
```

Comment:
Your status code is not correct, when someone tries to login with wrong password the http status code is not 404.

Reply:
I will definitely edit that thanks